server/game/player.js

```diff
@@ -497,7 +499,7 @@ class Player extends Spectator {
                 this.game.queueStep(new BestowPrompt(this.game, this, card));
             }
 
-            this.game.raiseMergedEvent('onCardEntersPlay', { card: card, playingType: playingType });
+            this.game.raiseMergedEvent('onCardEntersPlay', { card: card, playingType: playingType, originalLocation: originalLocation });
```

Comment:
The `onCardEntersPlay` event is raised a couple of other places, unfortunately. For consistency, you should update these usages as well:
`Game.takeControl` - probably the most important one since it directly involves characters.
`Player.attach` - will matter if they ever release a card that it matters how the attachment was placed.
`Player.flipPlotFaceup` - honestly, you can probably skip this one since it only relates to plots.